Refactor SearchParams to nuqs useQueryState

offbeatjs opened the pull request.

offbeatjs committed:
Refactor SearchParams to nuqs searchState

offbeatjs commented:
Check if this format is okayy, after that I'll go on implement nuqs to other files

max-programming commented:
@offbeatjs We'll need to see if it works on production as well. 

Can you please correctly implement it in one area that successfully works locally and in production?

offbeatjs committed:
Refactor repo filters to use nuqs query state

Replaces manual URLSearchParams and next/navigation search param handling with nuqs useQueryState and useQueryStates in pagination, sorter, and stars-filter components. Simplifies state management and improves consistency for query parameter updates across repo listing UI.

offbeatjs committed:
Update pagination.tsx

Resolve Conflicts

offbeatjs committed:
Merge branch 'main' into patch-2

offbeatjs commented:
Could you authorize the deployment ?

max-programming commented:
@offbeatjs It's gonna fail. You didn't install `nuqs`. I recommend making these changes locally rather than directly on GitHub

offbeatjs commented:
Could you clarify what you mean ? `nuqs:^2.7.2` is added to `package.json`

max-programming commented:
@offbeatjs This PR has your branch set to `patch-2` and `nuqs` is installed and implemented in **your** `main` branch

<img width="417" height="53" alt="image" src="https://github.com/user-attachments/assets/0ddfede1-8db9-4473-a2c3-b47cba3efb1e" />


Open a new PR **from your `main` branch**

max-programming commented:
@offbeatjs any updates?

offbeatjs commented:
Sorry,I was on a break, doing rn
